apps/openmw/mwmechanics/actors.cpp

```diff
@@ -555,12 +577,31 @@ namespace MWMechanics
             getRestorationPerHourOfSleep(ptr, health, magicka);
 
             DynamicStat<float> stat = stats.getHealth();
-            stat.setCurrent(stat.getCurrent() + health);
+            stat.setCurrent(stat.getCurrent() + health * hours);
             stats.setHealth(stat);
 
-            stat = stats.getMagicka();
-            stat.setCurrent(stat.getCurrent() + magicka);
-            stats.setMagicka(stat);
+            // We should take in account Stunted Magicka effects.
```

Comment:
```suggestion
            // Stunted Magicka effect should be taken into account.
```

It should be good to put all the stunted magicka stuff behind a check that the actor actually has an instance effect on them. :p